Added search ordering based on history.

moalhaddar opened the pull request:
Hello

This PR fixes a minor annoying thing that i face, each time i select an emoji it stays at the end of the search list.

This PR fixes that by maintaining the number of clicks of each emoji in sqlite and ordering based on that instead of the description length.

Regards,

FelipeFTN commented on emoji-copy@felipeftn/metadata.json:
```diff
@@ -6,5 +6,5 @@
   "settings-schema": "org.gnome.shell.extensions.emoji-copy",
   "gettext-domain": "emoji-copy",
-  "version": 21
+  "version": 22
```
The 23 version was already released! :sweat_smile:
Can you increase by one again to `23` pls? :pray: 
```suggestion
  "version": 23
```

FelipeFTN commented on emoji-copy@felipeftn/handlers/sql.js:
```diff
@@ -43,6 +43,12 @@ export class SQLite {
     `);
   }
 
+  increment_selection(unicode) {
+    this.query(
+      `UPDATE emojis SET clicked_times = clicked_times + 1 WHERE unicode = '${unicode}'`
+    )
+  }
```
Just little details to make the linter happy hahaha :smile: 
```suggestion
  increment_selection(unicode) {
    return this.query(`
      UPDATE emojis SET clicked_times = clicked_times + 1 WHERE unicode = '${unicode}'
    `);
  }
```

FelipeFTN commented on emoji-copy@felipeftn/handlers/sql.js:
```diff
@@ -87,6 +93,7 @@ export class SQLite {
       item.description = data[1];
       item.skin_tone = data[2];
       item.group = data[3];
+      item.clicked_times = data[4]
```
```suggestion
      item.clicked_times = data[4];
```

FelipeFTN reviewed (approved):
**Hello, @moalhaddar!! Thank you so much! What a nice contribution! :tada:**

I just added some pattern details, nothing big hahaha
**I tested the feature here, it's working great! Nice work!! :rocket:** 

I will take a time for you to apply the minor lint changes, or add some comments to the PR if you want, and I'll merge it! :people_hugging: 

**Nice work! Thanks!** :100:

moalhaddar commented:
Hello @FelipeFTN 

Thanks for the approval, i'll commit the comments soon.

I also have other minor bug fixes that i found, should i commit them in this same release or in another PR?

FelipeFTN commented:
**Nice!**
Don't worry, take your time with this PR! 💯 
As soon as you're ready to go, just let me know and I'll merge it right away! 🙏

moalhaddar commented:
I have added extra changes on this PR:
- addToClipboardAndStay was broken, now it works. Changing it to an arrow function keeps the reference of 'this' to the class.
- the `AndStay` functions family were triggering the paste if it's enabled. This is now fixed and they now stay.

FelipeFTN reviewed (approved):
**Thank you so much, buddy! :medal_sports:**

Your contribution will be available at the next Emoji Copy's release! :tada:
